File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.cpp
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    for (auto& num : arr) {
        if (num >= 50 && ((num & 1) == 0)) num /= 2;
        else if (num < 50 && ((num & 1) == 1)) num *= 2;
    }
    
    return arr;
}

File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        return Arrays
                .stream(arr)
                .map((x) -> (x >= 50 && ((x & 1) == 0)) ? (x / 2): ((x < 50 && ((x & 1) == 1)) ? (x * 2) : x)).toArray();
    }
}